tests/scripts/expect/posix-rcp-non-zero-iid.exp

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/expect -f
+#
+#  Copyright (c) 2023, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+# This test uses posix ot-cli and ot-rcp app with non-zero spinel iid support.
+# There are three nodes in this test (leader, rx-on-when-idle and rx-off-when-idle )
+# and each node spawn with non-zero iid.
+# This test perform energy scan, form and join the Thread network followed by ping
+# operation from a leader to each of the node.
+
+source "tests/scripts/expect/_common.exp"
+source "tests/scripts/expect/_multinode.exp"
+
+set is_mp_enabled $::env(MULTIPAN_RCP)
```

Comment:
I think that it is useful to have an option to skip running the test if it isn't configured to run but still executed; however, this command fails when I run on my machine without MULTIPAN_RCP defined:

```
no such variable  
    (read trace on "::env(MULTIPAN_RCP)")  
    invoked from within  
"set is_mp_enabled $::env(MULTIPAN_RCP)"  
    (file "./tests/scripts/expect/posix-rcp-non-zero-iid.exp" line 39)
```

For your consideration, this change on line 39 seems to work for me:

```
-set is_mp_enabled $::env(MULTIPAN_RCP)  
+set is_mp_enabled [info exists env(MULTIPAN_RCP)]
```

Reply:
On further inspection the environment variable is exported in `script/test` regardless of if MULTIPAN_RCP is enabled. The error that I see will still happen if the environment variable is not set, but there are several other variables that will cause similar errors, which implies that these tests are expected to be invoked by `script/test` or the environment variables must be manually set.